website_sale_product_brand/controllers/main.py

```diff
@@ -41,12 +23,12 @@ class WebsiteSale(website_sale):
                 website=True)
     def shop(self, page=0, category=None, brand=None, search='', **post):
         if brand:
-            request.context.setdefault('brand_id', int(brand))
-        result = super(WebsiteSale, self).shop(page=page, category=category,
-                                               brand=brand, search=search,
-                                               **post)
-        result.qcontext['brand'] = brand
-        return result
+            context = dict(request.env.context)
```

Comment:
we could use with_context() to update context values
**self = self.with_context(brand_id= int(brand))**